devDocs/startupShutdownInstall.md

```diff
@@ -0,0 +1,96 @@
+# NVDA Starting and Exiting outline
+
+## As a user, NVDA can be started by:
+
+1. For an installed copy:
+    1. Ctrl+Alt+N
+    1. Via secure desktop/ease of access
+    1. Via startup
+1. For an installed copy, portable copy, installer:
+    - By running the exe (see cli options too).
+    This can be triggered by a user or external process such as an existing NVDA instance
+    - test: `startupShutdownNVDA.Starts`
+1. For source: eg runnvda.bat
+
+## As a user, NVDA can be shutdown by:
+
+1. UI within NVDA, with and without an ExitDialog prompt (uses `triggerNVDAExit`):
+    1. NVDA+q
+        - test: `startupShutdownNVDA.Quits from keyboard, Restarts`
+    1. An unbound command to restart
+    1. A prompt to restart from changing certain settings
+    1. Via the NVDA menu -> Exit
+        - test: `startupShutdownNVDA.Quits from menu`
+1. A process sending `WM_QUIT`, eg a new NVDA process starting (uses `WM_QUIT`)
+1. A handled crash (directly causes a new process to start, terminates unsafely)
+    - test: `startupShutdownNVDA.Restarts on crash, Restarts on braille crash`
+1. An unhandled crash (terminates unsafely)
+1. An external command which kills the process (terminates unsafely) 
+1. Windows shutting down (terminates unsafely)
+    - On `wx.EVT_END_SESSION`, we save the config and play the exit sound. (Note: should this do more? NVDA may terminate unsafely directly after this)

```

Comment:
Windows does not give us very long to exit here. If we take too long we will be killed off anyway I believe. So saving config is the most important thing we should do. 
See https://docs.microsoft.com/en-us/windows/win32/shutdown/wm-endsession

Reply:
It seems like the urgency depends on the parameters passed in (eg the context of the shutdown). In my experience, apps can block a regular shutdown for as long they wish (Google Drive takes a while for me sometimes, unsaved notepads block indefinitely).

Reply:
The only difference for us is that in nvda.pyw we call `SetProcessShutdownParameters` which tells Windows to kill us off after all other apps. The reason we do this is so that if other apps take time to close, we can still actually as a screen reader read the Blocked Shutdown dialog and the list of apps, so that the NVDA user can accessibly go and manually shutdown those apps / answer any save dialogs etc. So, then with NVDA being killed off very last, if we hourselves seem to take too much time by Windows, then we ourselves may get included in the Block shutdown dialog, but the user won't be able to read it most probobally, as we would already be somewhere within our shutdown sequence.

Reply:
Is it worth adding this information to the docs?

```suggestion
    - On `wx.EVT_END_SESSION`, we save the config and play the exit sound.
        - Other actions are not performed as we have limited time to perform an action for this event. With NVDA being killed off very last, Windows may include NVDA in the Block shutdown dialog, but the user won't be able to read it if we are shutting down.
```